File: binding/src/libqhashfunctions.cpp
#include <QHashSeed>
#include <qhashfunctions.h>
#include "libqhashfunctions.h"
#include "libqhashfunctions.hxx"

QHashSeed* QHashSeed_new(const QHashSeed* other) {
	 return new QHashSeed(*other);
}

QHashSeed* QHashSeed_new2(QHashSeed* other) {
	 return new QHashSeed(*other);
}

QHashSeed* QHashSeed_new3() {
	 return new QHashSeed();
}

QHashSeed* QHashSeed_new4(unsigned long long d) {
	 return new QHashSeed(d);
}

void QHashSeed_CopyAssign(QHashSeed* self, QHashSeed* other) {
    *self = *other;
}

void QHashSeed_MoveAssign(QHashSeed* self, QHashSeed* other) {
    *self = std::move(*other);
}

unsigned long long QHashSeed_OperatorunsignedLongLong(const QHashSeed* self) {
	return self->operator unsigned long long();
}

QHashSeed* QHashSeed_GlobalSeed() {
	return new QHashSeed(QHashSeed::globalSeed());
}

void QHashSeed_SetDeterministicGlobalSeed() {
	QHashSeed::setDeterministicGlobalSeed();
}

void QHashSeed_ResetRandomGlobalSeed() {
	QHashSeed::resetRandomGlobalSeed();
}

void QHashSeed_Delete(QHashSeed* self) {
    delete self;
}

File: binding/include/libqhashfunctions.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qhashseed.html
pub const qhashseed = struct {

    /// New constructs a new QHashSeed object.
    pub fn New(other: ?*anyopaque) QtC.QHashSeed {
        return qtc.QHashSeed_new(@ptrCast(other));
    }


    /// New2 constructs a new QHashSeed object.
    pub fn New2(other: ?*anyopaque) QtC.QHashSeed {
        return qtc.QHashSeed_new2(@ptrCast(other));
    }


    /// New3 constructs a new QHashSeed object.
    pub fn New3() QtC.QHashSeed {
        return qtc.QHashSeed_new3();
    }


    /// New4 constructs a new QHashSeed object.
    pub fn New4(d: u64) QtC.QHashSeed {
        return qtc.QHashSeed_new4(d);
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QHashSeed_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QHashSeed_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn OperatorunsignedLongLong(self: ?*anyopaque, ) u64 {
        return qtc.QHashSeed_OperatorunsignedLongLong(@ptrCast(self));
    }

    pub fn GlobalSeed() QtC.QHashSeed {
        return qtc.QHashSeed_GlobalSeed();
    }

    pub fn SetDeterministicGlobalSeed() void {
        qtc.QHashSeed_SetDeterministicGlobalSeed();
    }

    pub fn ResetRandomGlobalSeed() void {
        qtc.QHashSeed_ResetRandomGlobalSeed();
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QHashSeed_Delete(@ptrCast(self));
    }
};
